### ISSUE TITLE
dbg-macro should tolerate use of dbg! in tests

### ISSUE THREAD

**Issue by ijackson** [rust-lang/rust-clippy#8481]
### Summary

`cargo clippy -- -D dbg-macro` complains about use of `dbg!` even in functions marked `#[test]`.  But this is a correct use.  As the docs for the macro have it

> Note that the macro is intended as a debugging tool and therefore you should avoid having uses of it in version control for long periods (**other than in tests and similar**). Debug output from production code is better done with other facilities ...

(emphasis mine).  In a `#[test]` function. the `dbg!` output is properly captured and only appears if the test fails or with `--show-output`.  IMO `dbg!` in a test is a good approach.  `dbg!` should be tolerated in any code which is `#[cfg(test)]`.

### Lint Name

dbg-macro

### Reproducer

I tried this code:

```rust
fn main() {
}

#[test]
fn wombat() {
    dbg!("wombat");
}
```

I saw this happen:

```
zealot:experiments> nailing-cargo +stable clippy -- -D dbg-macro
nailing-cargo: out-of-tree, building in: `/home/ian/Rustup/Arti/Build/experiments'
nailing-cargo: using really to run as user `rustcargo'
nailing-cargo: *WARNING* cwd is not in Cargo.nail thbough it has Cargo.toml!
nailing-cargo: nailed (0 manifests, 0 packages)
nailing-cargo: invoking: cargo +stable clippy --manifest-path=/home/ian/Rustup/Arti/experiments/Cargo.toml --locked --target-dir=target --offline -- -D dbg-macro
    Checking foo v0.1.0 (/home/ian/Rustup/Arti/experiments)
warning: lint name `dbg_macro` is deprecated and does not have an effect anymore. Use: clippy::dbg_macro
  |
  = note: requested on the command line with `-D dbg_macro`

warning: `foo` (bin "foo") generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.16s
nailing-cargo: unnailed.  status 0.
zealot:experiments> 
```

I expected to see this happen:

No complaint.

### Version

```text
rustcargo@zealot:/volatile/rustcargo/Rustup/Arti/experiments$ rustc +stable -Vv
rustc 1.59.0 (9d1b2106e 2022-02-23)
binary: rustc
commit-hash: 9d1b2106e23b1abd32fce1f17267604a5102f57a
commit-date: 2022-02-23
host: x86_64-unknown-linux-gnu
release: 1.59.0
LLVM version: 13.0.0
rustcargo@zealot:/volatile/rustcargo/Rustup/Arti/experiments$
```
